functions/common/datastore.js

```diff
@@ -66,28 +68,22 @@ class Datastore {
 
   fetchForm(projectId, featureTypeId, formId) {
     return this.fetchDoc_(
-      `projects/${projectId}/featureTypes/${featureTypeId}/forms/${formId}`);
+      `projects/${projectId}/featureTypes/${featureTypeId}/forms/${formId}`
+    );
   }
 
   fetchSheetsConfig(projectId) {
     return this.fetchDoc_(`projects/${projectId}/sheets/config`);
   }
 
-  async insertFeature(projectId, layerId, { caption, location }) {
-    const docRef = await this.db_.collection('projects').doc(projectId);
-    if (!docRef.exists) {
-      console.log("Project " + projectId + " does not exist!");
-      // For debug purposes uncomment below to create the project
-      // docRef.set({ project: projectId });
-      throw new Error(`Project ${projectId} does not exist!`);
-      return;
+  async insertFeature(projectId, feature) {
+    const docRef = await this.db_.collection("projects").doc(projectId);
+    const doc = await docRef.get();
+    if (!doc.exists) {
+      throw new Error(`/projects/${projectId} not found`);
     }
-    await docRef.collection('features').add({
-      layerId: layerId,
-      caption: caption,
-      location: location
-    });
+    await docRef.collection("features").add(feature);
   }
 }
 
-module.exports = Datastore;
\ No newline at end of file
+module.exports = Datastore;
```

Comment:
This is identical as far as I see!

Reply:
There were a few changes to `insertFeature()`. Sorry for the formatting change!